cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/write_cloudwatch_agent_json.py

```diff
@@ -132,6 +137,44 @@ def select_logs(configs, args):
     return selected_configs
 
 
+def get_node_roles(scheudler_plugin_node_roles):
+    node_type_roles_map = {"ALL": ["ComputeFleet", "HeadNode"], "HEAD": ["HeadNode"], "COMPUTE": ["ComputeFleet"]}
+    return node_type_roles_map.get(scheudler_plugin_node_roles)
+
+
+def load_config(cluster_config_path):
+    with open(cluster_config_path) as input_file:
+        return yaml.load(input_file, Loader=yaml.SafeLoader)
+
+
+def add_scheduler_plugin_log(config_data, cluster_config_path):
+    """Add custom log files to config data if log files specified in scheduler plugin."""
+    cluster_config = load_config(cluster_config_path)
+    if (
+            get_dict_value(cluster_config, "Scheduling.SchedulerSettings.SchedulerDefinition.Monitoring.Logs.Files")
+            and get_dict_value(cluster_config, "Scheduling.Scheduler") == "plugin"
+    ):
+        log_files = get_dict_value(
+            cluster_config, "Scheduling.SchedulerSettings.SchedulerDefinition.Monitoring.Logs.Files"
+        )
+        for i in range(len(log_files)):
+            # Add log config
+            log_config = {
+                "timestamp_format_key": "scheduler_plugin_log_" + str(i),
+                "file_path": log_files[i].get("FilePath"),
+                "log_stream_name": log_files[i].get("FilePath"),
```

Comment:
we shuold probably use the file name rather than the full path

Reply:
After adding `LogStreamName` as mandatory in the config as we discuss, jere I changed to `"timestamp_format_key": "scheduler_plugin_log_" + str(i)` to use `LogStreamName`